src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -18595,14 +18912,14 @@ static void F(bool b, B<object?> x, B<object> y)
     {
         var z = Create(A.F); // B<object~>
         object o;
-        o = (new[] { x, y })[0]/*T:B<object!>!*/;
+        o = (new[] { x, y })[0]/*T:B<object!>*/;
```

Comment:
> o = (new[] { x, y })[0]/*T:B<object!>*/; [](start = 7, length = 41)

This change feels unexpected. Both inputs are not nullable at the top level, result shouldn't be oblivious

Reply:
If we can't infer a best type, then we don't know how to convert, so we don't know what the top-level nullability is. 
We could make some assumptions, but I ran into some problems going down that route. Probably solvable though.